### PR TITLE
{math}[foss/2018b] XMDS2 2.2.3 (REVIEW)

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

File: easybuild/easyconfigs/x/XMDS2/XMDS2-2.2.3-foss-2018b-Python-2.7.15.eb [hunk 1]
```diff
@@ -0,0 +1,64 @@
+easyblock = 'Bundle'
```
Comment (boegel) [inline easybuild/easyconfigs/x/XMDS2/XMDS2-2.2.3-foss-2018b-Python-2.7.15.eb]: @hajgato Do you mind changing this to the new `PythonBundle`, which will be included in EasyBuild v3.8.0 (cfr. https://github.com/easybuilders/easybuild-easyblocks/pull/1553)?

File: easybuild/easyconfigs/x/XMDS2/XMDS2-2.2.3-foss-2018b-Python-2.7.15.eb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+version = '2.2.3'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'http://www.xmds.org/'
+description = """ The purpose of XMDS2 is to simplify the process of creating simulations that
+ solve systems of initial-value first-order partial and ordinary differential equations."""
+
+toolchain = {'name': 'foss', 'version': '2018b'}
+
+
```
Comment (boegel) [inline easybuild/easyconfigs/x/XMDS2/XMDS2-2.2.3-foss-2018b-Python-2.7.15.eb]: @hajgato Style nitpicking: please remove duplicate empty line

File: easybuild/easyconfigs/x/XMDS2/XMDS2-2.2.3-foss-2018b-Python-2.7.15.eb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    ('lxml', '4.2.5', versionsuffix),
+    ('mpmath', '1.0.0', versionsuffix),
+    ('h5py', '2.8.0', versionsuffix),
+]
+
+exts_defaultclass = 'PythonPackage'
+exts_default_options = {
+    'download_dep_fail': True,
+    'use_pip': True,
+}
```
Comment (boegel) [inline easybuild/easyconfigs/x/XMDS2/XMDS2-2.2.3-foss-2018b-Python-2.7.15.eb]: This whole block can be replaced with just `use_pip = True` if the `PythonBundle` easyblock is used

File: easybuild/easyconfigs/x/XMDS2/XMDS2-2.2.3-foss-2018b-Python-2.7.15.eb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        'preinstallopts': 'make &&'
+    }),
+]
+
+sanity_check_paths = {
+    'files': ['bin/xmds2'],
+    'dirs': ['lib/python%(pyshortver)s/site-packages'],
+}
+
+modextrapaths = {'PYTHONPATH': ['lib/python%(pyshortver)s/site-packages']}
```
Comment (boegel) [inline easybuild/easyconfigs/x/XMDS2/XMDS2-2.2.3-foss-2018b-Python-2.7.15.eb]: this line can be removed if the `PythonBundle` easyblock is used

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 3 (3 easyconfigs in this PR)
node2060.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/3343621de370f7c279a7c1292156f4cd for a full test report.

**Comment by boegel** [comment]
@hajgato (still) a checksum issue?

**Comment by hajgato** [comment]
@boegel its sourceforge-download issue. For some reason, only manual download works.

**Comment by hajgato** [comment]
@boegel changed to `PythonBundle`

File: easybuild/easyconfigs/x/XMDS2/XMDS2-2.2.3-foss-2018b-Python-2.7.15.eb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        'checksums': ['f353aab21fd474459d97b709e527b5571314ee5f067441dc9f88e33eecd96592'],
+    }),
+    ('Markdown', '3.0.1', {
+        'source_urls': ['https://pypi.python.org/packages/source/m/Markdown/'],
+        'checksums': ['d02e0f9b04c500cde6637c11ad7c72671f359b87b9fe924b2383649d8841db7c'],
+    }),
+    ('xmds', version, {
+        'modulename': 'xpdeint',
+        'patches': ['XMDS2-2.2.3_HDF5_OpenBLAS.patch'],
+        'source_urls': ['http://download.sourceforge.net/projects/xmds'],
```
[the remark this answers is not in the document]
Comment (hajgato) [reply easybuild/easyconfigs/x/XMDS2/XMDS2-2.2.3-foss-2018b-Python-2.7.15.eb]: @boegel fixed.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2417.golett.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/3f2e441c4e58ea10e110377276f224eb for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node3100.skitty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/54bb1d12cd56b609d7818eef76be99e7 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2060.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/48510be65c41564bd380d963498f25f4 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @hajgato!

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2104.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/92bb645a27868981e6fd5e1c475e4de9 for a full test report.

**Pull request merged**